framework/Optimizers/GradientBasedOptimizer.py

```diff
@@ -471,17 +505,17 @@ def localFinalizeActualSampling(self,jobObject,model,myInput):
             if self.convergeTraj[traj]:
               self.status[traj] = {'process':None, 'reason':'converged'}
             else:
-              # if rejecting bad point, keep the old point as the new point
-              if self.status[traj]['reason'] != 'rejecting bad opt point':
-                try:
-                  self.counter['recentOptHist'][traj][1] = copy.deepcopy(self.counter['recentOptHist'][traj][0])
-                except KeyError:
-                  # this means we don't have an entry for this trajectory yet, so don't copy anything
-                  pass
-                self.counter['recentOptHist'][traj][0] = {'inputs':self.optVarsHist[traj][self.counter['varsUpdate'][traj]],
-                                                          'output':currentObjectiveValue}
               # update status to submitting grad eval points
               self.status[traj]['process'] = 'submitting grad eval points'
+            # if rejecting bad point, keep the old point as the new point; otherwise, add the new one
+            if self.status[traj]['reason'] != 'rejecting bad opt point':
```

Comment:
Using comparison to a string to determine if the bad point was found, is not the best practice (somewhat expensive computationally, and easy to change in one place and forget to change in another.)